### ISSUE TITLE
CA1721 (Property names should not match get methods) fires on every type that has a property named Type

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn-analyzers#1374]
#### Analyzer package

Microsoft.CodeQuality.Analyzers

#### Analyzer

PropertyNamesShouldNotMatchGetMethodsAnalyzer

#### Repro steps

```
class T { }
class C
{
    public T Type { get; }
}
```

#### Expected behavior
No CA1721. FxCop does not fire for this case.

#### Actual behavior
`warning CA1721: The property name 'Type' is confusing given the existence of method 'GetType'. Rename or remove one of these members.`

We should ignore `Object.GetType` in our analysis. Otherwise, this rule is just noise.

**Comment by sharwell** [comment]
I'm undecided here. Clearly the situation falls into the category this rule was intended to detect. The possibly-redeeming factor is enough people have seen `GetType` to not get it confused.

**Comment by 333fred** [comment]
Having a property named `Type` is pretty common. Roslyn uses it extensively, for example. I'm in favor.

**Comment by thomaslevesque** [comment]
This seems to have been fixed almost 3 years ago, but I'm still seeing this warning with a `Type` property... Regression?
I'm using Microsoft.CodeAnalysis.FxCopAnalyzers 3.0.0.

**Comment by mavasani** [comment]
@thomaslevesque Can you check if you have "Microsoft CodeAnalysis 2019" VSIX extension installed? If so, please uninstall it and retry.

**Comment by thomaslevesque** [comment]
No, I don't. I'm not using Visual Studio, just building from the command line.

But I have some additional information: it seems to happen only in classes that inherit `System.Exception`.

```csharp
    public class Foo
    {
        public Foo(string type) => Type = type;

        // No warning
        public string Type { get; }
    }

    #pragma warning disable CA1032 // to reduce noise
    public class FooException : Exception
    {
        public FooException(string type, string message)
            : base(message) => Type = type;

        // CA1721
        public string Type { get; }
    }
```

**Comment by thomaslevesque** [comment]
Ah, I see what the problem is. [`System.Exception` hides `GetType`](https://github.com/dotnet/runtime/blob/b1826278d4dda34b41d438ad832ac8f267458ef4/src/libraries/System.Private.CoreLib/src/System/Exception.cs#L197):

```csharp
        // this method is required so Object.GetType is not made virtual by the compiler
        // _Exception.GetType()
        public new Type GetType() => base.GetType();
```

So when the analyzer checks the method like this:
https://github.com/dotnet/roslyn-analyzers/blob/39d189ab8b48b3811978edbafdc0aef190025f53/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/PropertyNamesShouldNotMatchGetMethods.cs#L106

It sees `System.Exception` as the `ContainingType`, not `System.Object`.

Should `System.Exception` be treated as a special case?
